### PR TITLE
feat(auth): Return basic user info in OIDC ID token

### PR DESCRIPTION
**Pull request by EricHasegawa**
Returns basic user info inside the ID token we issue from the `/token` endpoint

### CODE REVIEW

File: src/sentry/web/frontend/oauth_token.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def _get_open_id_token(self, grant, request):
                 secrets.token_urlsafe(),
                 nonce=request.POST.get("nonce"),
             )
-            return open_id_token.get_encrypted_id_token()
+            return open_id_token.get_encrypted_id_token(grant)
```
Comment (nhsiehgit) [inline src/sentry/web/frontend/oauth_token.py:93]: kwarg?

File: src/sentry/web/frontend/openidtoken.py [hunk 2]
```diff
@@ -47,6 +48,24 @@ def get_encrypted_id_token(self):
             "exp": self.exp,
             "iat": self.iat,
         }
+        user_details = self._get_user_details(grant)
```
Comment (nhsiehgit) [inline src/sentry/web/frontend/openidtoken.py:51]: kwarg

File: src/sentry/web/frontend/openidtoken.py [hunk 3]
```diff
@@ -47,6 +48,24 @@ def get_encrypted_id_token(self):
...
+        claims.update(user_details)
         if self.nonce:
             claims["nonce"] = self.nonce
         return jwt_utils.encode(claims, self.shared_secret, headers={**headers, "alg": "HS256"})
+
+    def _get_user_details(self, grant):
```
Comment (nhsiehgit) [inline src/sentry/web/frontend/openidtoken.py:57]: add types/return types?

File: src/sentry/web/frontend/openidtoken.py [hunk 4]
```diff
@@ -35,7 +36,7 @@ def __init__(
         self.exp = exp if exp else default_expiration()
         self.iat = iat if iat else timezone.now()
 
-    def get_encrypted_id_token(self):
+    def get_encrypted_id_token(self, grant):
```
Comment (nhsiehgit) [inline src/sentry/web/frontend/openidtoken.py:39]: types/return types?

**Review by nhsiehgit — COMMENTED** [review]
typing nits

File: src/sentry/web/frontend/openidtoken.py [hunk 5]
```diff
@@ -47,6 +48,24 @@ def get_encrypted_id_token(self):
...
+            profile_details = {
+                "name": grant.user.name,
+                "avatar_type": grant.user.avatar_type,
+                "avatar_url": grant.user.avatar_url,
+                "date_joined": str(grant.user.date_joined),
+            }
+            user_details.update(profile_details)
+        if grant.has_scope("email"):
+            email = UserEmail.objects.get(user=grant.user)
+            email_details = {"email": email.email, "email_verified": email.is_verified}
```
Comment (markstory) [inline src/sentry/web/frontend/openidtoken.py:69]: Users can have *many* email addresses. Using `.get()` here will fail for lots of folks. You should have a list of email addresses, or only show the 'primary' email address which is in user.email.

**Review by JoshFerge — DISMISSED** [review]
can we add the security team as a codeowner of these files?

File: src/sentry/web/frontend/openidtoken.py (hunk 5 above)
Comment (EricHasegawa) [reply src/sentry/web/frontend/openidtoken.py]: Good catch! Didn't think about this - fixed it

**Comment by EricHasegawa** [comment]
> can we add the security team as a codeowner of these files?

Addressed [here](https://github.com/getsentry/sentry/pull/52761)

**Review by JoshFerge — COMMENTED** [review]
looks good from the codeowners perspective. would look at the user email stuff as mark pointed out.

**Review by markstory — APPROVED** [review]

**Pull request merged**